File: src/test/java/io/gluth/pagespace/domain/LinkTest.java
package io.gluth.pagespace.domain;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class LinkTest {

    private final Page physics = new Page("physics", "Physics");
    private final Page math    = new Page("math", "Mathematics");

    @Test
    void storesSourceAndTarget() {
        Link link = new Link(physics, math);
        assertEquals(physics, link.source());
        assertEquals(math, link.target());
    }

    @Test
    void equalByEndpoints() {
        Link a = new Link(physics, math);
        Link b = new Link(physics, math);
        assertEquals(a, b);
        assertEquals(a.hashCode(), b.hashCode());
    }

    @Test
    void notEqualForDifferentEndpoints() {
        Link a = new Link(physics, math);
        Link b = new Link(math, physics);
        assertNotEquals(a, b);
    }

    @Test
    void nullSourceThrows() {
        assertThrows(NullPointerException.class, () -> new Link(null, math));
    }

    @Test
    void nullTargetThrows() {
        assertThrows(NullPointerException.class, () -> new Link(physics, null));
    }
}

File: src/test/kotlin/io/gluth/pagespace/domain/LinkTest.kt
package io.gluth.pagespace.domain

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows
import kotlin.test.assertEquals
import kotlin.test.assertNotEquals

class LinkTest {

    private val physics = Page("physics", "Physics")
    private val math    = Page("math", "Mathematics")

    @Test
    fun storesSourceAndTarget() {
        val link = Link(physics, math)
        assertEquals(physics, link.source)
        assertEquals(math, link.target)
    }

    @Test
    fun equalByEndpoints() {
        val a = Link(physics, math)
        val b = Link(physics, math)
        assertEquals(a, b)
        assertEquals(a.hashCode(), b.hashCode())
    }

    @Test
    fun notEqualForDifferentEndpoints() {
        val a = Link(physics, math)
        val b = Link(math, physics)
        assertNotEquals(a, b)
    }

    @Test
    fun nullSourceThrows() {
        assertThrows<NullPointerException> { Link(null, math) }
    }

    @Test
    fun nullTargetThrows() {
        assertThrows<NullPointerException> { Link(physics, null) }
    }
}
